src/Silex/Provider/HttpKernelServiceProvider.php

```diff
@@ -27,8 +35,29 @@ public function register(Container $app)
             return new ControllerResolver($app, $app['logger']);
         };
 
+        $app['argument_metadata_factory'] = function ($app) {
+            return new ArgumentMetadataFactory();
+        };
+        $app['argument_value_resolvers'] = function ($app) {
+            return array(
+                new RequestAttributeValueResolver(),
```

Comment:
You services will still be registered when using old Symfony versions, but will just be unusable. I'd recommend that you do something like this instead:
```php
if (Kernel::VERSION_ID >= 30100) {
        $app['argument_metadata_factory'] = function ($app) {
            return new ArgumentMetadataFactory();
        };
        $app['argument_value_resolvers'] = function ($app) {
            return array(...);
        };
        $app['argument_resolver'] = function ($app) {
            return new ArgumentResolver($app['argument_metadata_factory'], $app['argument_value_resolvers']);
        };
}
$app['kernel'] = function ($app) {
    return new HttpKernel($app['dispatcher'], $app['resolver'], $app['request_stack'], isset($app['argument_resolver']) ? $app['argument_resolver'] : null);
};
```